sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -172,4 +173,15 @@ class DefaultSource extends FileFormat with DataSourceRegister {
         .mapPartitions(_.map(pair => new String(pair._2.getBytes, 0, pair._2.getLength, charset)))
     }
   }
+
+  private def verifySchema(schema: StructType): Unit = {
+    schema.foreach { field =>
+      field.dataType match {
+        case _: ArrayType | _: MapType | _: StructType =>
+          throw new AnalysisException(
```

Comment:
Just for a note, I am less sure of `AnalysisException`. It seems JSON uses `sys.error()`. CSV data source uses `UnsupportedOperationException`, `RuntimeException` and etc. Maybe we need to make the exceptions consistent later.

It seems `AnalysisException` is 

>Thrown when a query fails to analyze, usually because the query itself is invalid.

So, I think it would be nicer to be consistent with CSV existing behaviour or JSON one. FYI, CSV is throwing `RuntimeException` when it fails to read due to types.